guides/v2.2/javascript-dev-guide/widgets/widget-remaining-characters.md

```diff
@@ -0,0 +1,150 @@
+---
+group: javascript-developer-guide
+subgroup: 3_Widgets
+title: RemainingCharacters widget
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+---
+
+The RemainingCharacters [widget](https://glossary.magento.com/widget/) allows you to add the remaining characters note for a text field.
+
+The RemainingCharacters widget can be used only on the frontend area.
+
+The RemainingCharacters widget source is [`<Magento_Catalog_module_dir>/view/frontend/web/js/product/remaining-characters.js`][].
+
+## Initialize the RemainingCharacters widget
+
+For information about how to initialize a widget in a JS component or `.phtml` template, see the [Initialize JavaScript][] topic.
+
+The RemainingCharacters widget is instantiated with:
+
+```javascript
+$("#remaining-characters").remainingCharacters({
+    maxLength: 5,
+    noteSelector: '.note',
+    counterSelector: '.note .character-counter'
+});``
+```
+
+Where:
+
+- `#remaining-characters` is the selector of the element which will be the RemainingCharacters.
+
+The following example shows a PHTML file using the script:
+
+```html
+<script>
+    require([
+        'jquery',
+        'Magento_Catalog/js/product/remaining-characters'
+    ], function ($) {
+        'use strict';
+
+        $("#remaining-characters").remainingCharacters({
+            maxLength: 5,
+            noteSelector: '.note',
+            counterSelector: '.note .character-counter'
+        });
+    });
+</script>
+```
+
+## Options
+
+- [counterSelector](#counterselector)
+- [errorClass](#errorclass)
+- [maxLength](#maxlength)
+- [noDisplayClass](#nodisplayclass)
+- [noteSelector](#noteselector)
+- [remainingText](#remainingtext)
+- [tooManyText](#toomanytext)
+
+### `counterSelector`
+
+The selector of counter element.
+
+**Type**: String
+
+**Default value**: `undefined`
+
+### `errorClass`
+
+The error class that appends to the [note element](#noteselector) if the [maxLength](#maxlength) is exceeded.
+
+**Type**: String
+
+**Default value**: `'mage-error'`
+
+### `maxLength`
+
+The maximum length of the text for the field.
+
+**Type**: Integer
+
+**Default value**: `undefined`
+
+### `noDisplayClass`
+
+The class that appends to the [counter element](#counterselector) if the field value is empty.
+
+**Type**: String
+
+**Default value**: `'no-display'`
+
+### `noteSelector`
+
+The selector of note element.
```

Comment:
Hi @serhiyzhovnir. Thank you for the great topic!
Could you add more info about the note element, please? At first glance, it's not completely clear what's the note component and its main purpose.

Thank you!